### PR TITLE
Add GDAL 2.2.0 to Travis build matrix

### PR DESCRIPTION
**Pull request by geowurster**
Released yesterday: http://download.osgeo.org/gdal/2.2.0/

### CODE REVIEW

**Comment by snorfalorpagus** [comment]
I tried restarting the Travis build, but it's still falling over on something related to `pip` and the `six` module:

https://travis-ci.org/Toblerity/Fiona/jobs/231022585

**Comment by rbuffat** [comment]
@snorfalorpagus looks like this bug here: https://github.com/pypa/setuptools/issues/951

**Comment by sgillies** [comment]
I can't tell whether pinning setuptools back to 33.0 will fix this or not, but I'll give it a try.

Nevertheless, I see in one test failure that the default name of a GeoJSON file's single layer is no longer 'OgrGeoJSON'. Seems to be the only problem :)

**Comment by micahcochran** [comment]
Could this be closed and reopened to retest the PR?

**Comment by snorfalorpagus** [comment]
The previous commit fixes the existing errors related to GeoJSON. I'm working on a fix for #460.

**Comment by snorfalorpagus** [comment]
I can't figure out how to get Cython to compile something that works in both GDAL < 2.2 and GDAL >= 2.2 at the same time. I tried using the compile time directives but they don't seem to work the way I expected. If we can figure this out it'll be the same answer for how to merge ogrext1 and ogrext2 into a single source. Can anyone else figure this out?

**Comment by snorfalorpagus** [comment]
This is working now. It was a stupid mistake in checking the version numbers.

**Comment by geowurster** [comment]
@snorfalorpagus Thanks for picking this up!

File: fiona/ogrext2.pyx [hunk 1]
```diff
@@ -35,6 +35,10 @@ from cpython cimport PyBytes_FromStringAndSize, PyBytes_AsString
 
 
 log = logging.getLogger("Fiona")
+class NullHandler(logging.Handler):
+    def emit(self, record):
+        pass
+log.addHandler(NullHandler())
```
Comment (snorfalorpagus) [inline fiona/ogrext2.pyx]: Not sure if this should be here? I was a little confused with the rebase.

Comment (sgillies) [inline fiona/ogrext2.pyx]: Yes, it belongs. The way I did logging early on was ignorant of how Python's hierarchical logger works and is wrong. In the future we should remove all these `NullHandler` classes, but keep it for now.

File: tests/test_feature.py [hunk 2]
```diff
@@ -110,3 +111,18 @@ def test_properties(self):
         g = featureRT(f, self.c)
         self.assertEqual(g['properties']['title'], 'foo')
 
+def test_feature_null_field(tmpdir):
+    """
+    In GDAL 2.2 the behaviour of OGR_F_IsFieldSet slightly changed.
+    See GH #460.
```
Comment (sgillies) [inline tests/test_feature.py]: Fantastic, @snorfalorpagus!

**Pull request merged**